Assets/Feature01/Scripts/CharacterStates/PolarityManager.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PolarityManager : MonoBehaviour
+{
+
+
+    public float MaxHealth;
+    public float CurrentHealth;
+    public float MovementSpeed;
+    public Color PlayerColor;
+    public Stats CurrentPlayerStats;
+    public PlayerState CurrentPlayerState;
+
+    [SerializeField] ShootSystem ShootSystem;
+    [SerializeField] PlayerController PlayerController;
+
+
+    Dictionary<PlayerState, Stats> PlayerStatesSelection = new Dictionary<PlayerState, Stats>();
+    
+   
+
+
+    private void Awake()
+    {
+        PlayerStatesSelection =  new Dictionary<PlayerState, Stats>()
+        {
+        {PlayerState.Ninja, Resources.Load<Stats>("ScriptableObjects/Ninja") },
+        {PlayerState.Tank, Resources.Load<Stats>("ScriptableObjects/Tank") },
+        {PlayerState.General, Resources.Load<Stats>("ScriptableObjects/General")}
+        };
+
+        CurrentPlayerStats = PlayerStatesSelection[PlayerState.General];
+        CurrentPlayerState = PlayerState.General;
+
+        MaxHealth = CurrentPlayerStats.MaxHealth;
+        MovementSpeed = CurrentPlayerStats.MovementSpeed;
+        CurrentHealth = 100f;
```

Comment:
You should probably have a method to set the health. As you know the max health is a limiting condition. What happens if max hp is 75 and you are setting 100f ?

Reply:
You should probably do this from the health manager class